sentry/src/db/channel.rs

```diff
@@ -67,3 +82,93 @@ pub async fn insert_channel(
         })
         .await
 }
+
+pub async fn list_channels(
+    pool: &DbPool,
+    skip: u64,
+    limit: u32,
+    creator: &Option<String>,
+    validator: &Option<ValidatorId>,
+    valid_until_ge: &DateTime<Utc>,
+) -> Result<Vec<Channel>, RunError<bb8_postgres::tokio_postgres::Error>> {
+    let validator = validator.as_ref().map(|validator_id| {
+        serde_json::Value::from_str(&format!(r#"[{{"id": "{}"}}]"#, validator_id))
+            .expect("Not a valid json")
+    });
+    let (where_clauses, params) =
+        channel_list_query_params(creator, validator.as_ref(), valid_until_ge);
+
+    pool
+        .run(move |connection| {
+            async move {
+                // To understand why we use Order by, see Postgres Documentation: https://www.postgresql.org/docs/8.1/queries-limit.html
+                let statement = format!("SELECT id, creator, deposit_asset, deposit_amount, valid_until, spec FROM channels WHERE {} ORDER BY id DESC LIMIT {} OFFSET {}", where_clauses.join(" AND "), limit, skip);
+                match connection.prepare(&statement).await {
+                    Ok(stmt) => {
+                        match connection.query(&stmt, params.as_slice()).await {
+                            Ok(rows) => {
+                                let channels = rows.iter().map(Channel::from).collect();
+
+                                Ok((channels, connection))
+                            },
+                            Err(e) => Err((e, connection)),
+                        }
+                    },
+                    Err(e) => Err((e, connection)),
+                }
+            }
+        })
+        .await
+}
+
+pub async fn list_channels_total_pages(
```

Comment:
Just realized that this is counting all records, which means that we need to calculate the TotalPages in the route handler since it should be calculated accordingly to the limit per page.

Reply:
This is fixed now.